lib/generators/solidus_paypal_commerce_platform/install/install_generator.rb

```diff
@@ -3,16 +3,15 @@
 module SolidusPaypalCommercePlatform
   module Generators
     class InstallGenerator < Rails::Generators::Base
-      class_option :auto_run_migrations, type: :boolean, default: false
-
-      # This should only be used by the solidus installer prior to v3.3.
-      class_option :skip_migrations, type: :boolean, default: false, hide: true
+      class_option :migrate, type: :boolean, default: false
+      class_option :specs, type: :string, enum: %w[all frontend], default: 'frontend'
```

Comment:
Do you think it's even worth it to provide the `all` option?

Reply:
I see. What do you think about modifying `bin/rspec` to do two things:

- Run the `frontend` specs within the dummy app.
- Run the rest of the regular specs from the extension root directory.

In my view, it communicates better what we want to do, as it doesn't make sense to test the code of a dependency from within the application.

Reply:
@waiting-for-dev this would also be my preference, but I'm still concerned if that's the right approach because:

1. We will end up having two different approaches for testing things. This means having two different paths to maintain, two different spec_helpers and dependencies to maintain (we are relying on the spec_helper and test dependencies added with the SSF now), and more importantly two paths to document and explain to people that want to contribute.
2. I've been thinking if there are possibilities of having feature specs that involve both frontend and backend, like a full journey from changing an admin preference to how that is reflected in the storefront. In that case, it would be impossible.

Reply:
@kennyadsl, I see your point. But I think that's what makes sense. In the end, if we need two different approaches is because they're two different things. I don't think it'll help us hide that complexity under the carpet. On the contrary, blurring that distinction in development will make it too easy to cross boundaries and regret if there's a moment where we find a more modular approach to the frontend. Think of it as mixing two different libraries in a single one. If we decide to extract the SSF code to another repo at one point, ideally, it should only be a matter of moving files around.

Reply:
@waiting-for-dev @kennyadsl I agree that having two different paths would better mirror the two different approaches we have for core+backend vs frontend, and I would be in favor of that if it wasn't for these other facets of the discussion:

1. we actually wanted to copy over all specs from extensions in cometeer to have more controlled upgrades and ensuring our monkey patches were not breaking other features of the extensions (namely we did this for solidus_subscriptions and solidus_stripe IIRC).
2. the burden of maintaining a different path seems to much for me, especially considering that the "old way" was also relying on a very contrived dummy app on which we ran specs, the only difference is that it re-used the same Gemfile and the call-site of rspec was the extension root.

Would this be ok from the above point of view? Other aspects we're missing?

Reply:
> we actually wanted to copy over all specs from extensions in cometeer to have more controlled upgrades and ensuring our monkey patches were not breaking other features of the extensions (namely we did this for solidus_subscriptions and solidus_stripe IIRC).

For me, this is still a partially weak argument. Being the store not owning that code (it's a dependency that will receive updates), copying the whole test suite will generate a lot of maintenance work when upgrading the extension (people have to check manually which specs changed from version to version). I understand that someone wants that, but IMHO can't be the default.

 > the burden of maintaining a different path seems to much for me, especially considering that the "old way" was also relying on a very contrived dummy app on which we ran specs, the only difference is that it re-used the same Gemfile and the call-site of rspec was the extension root.

Right, we were still running specs on a generated dummy app (generated with the `test_app` rake task). It is not drastically different from what we are doing now, we are just copying the specs over that application, instead of running them from the engine with its gemset. 

I still can't see my preference, though. Both approaches have pros and cons. @solidusio/core-team do you have any opinion here?

Reply:
> but IMHO can't be the default.

Just for clarity, this is not the default, installing this extension through the installer will only copy over system/frontend specs, the `all` option must be explicitly set calling the installer manually and also supports the extension development.

Reply:
> the burden of maintaining a different path seems to much for me, especially considering that the "old way" was also relying on a very contrived dummy app on which we ran specs. The only difference is that it re-used the same Gemfile and the call-site of rspec was the extension root.

You're right, but those two differences are essential in my view. The old setup was the regular one for any library: using the library Gemfile & root directory to run tests. I understand that the dummy app was only used as support to test integration tests (which has now been integrated into SSF), but unit tests didn't use that.

Reply:
> I understand that the dummy app was only used as support to test integration tests (which has now been integrated into SSF), but unit tests didn't use that.

This is true only partially, the space in which any extension is moving is that of Rails engines/railties, so the initialization process of rails is required for testing anything inside `app/` and inside `lib/` if it interacts in any way with models or Solidus core.

I fully get the point of being "surprised" by an unusual setup (but it will be usual once we start using it in our ecosystem), but don't fully get how the difference is essential (I see it as very blurry), can you please expand on it?

Reply:
Hey, @elia. When you test a Rails model at the unit level, you only require its file, but you don't need to go through all Rails initialization processes. That should also be the case for our models inheriting Solidus ones. Does it make sense?

However, my point is that we're thinking about it as a temporary solution, so we shouldn't make the process consolidated to a point where we remove the line in our development pipeline, making it too easy to make the wrong assumptions and 1) breaking apps not using SSF & 2) making our life harder once we come with a definitive solution (like the solidus_checkout extension you proposed).

Reply:
> @waiting-for-dev what I meant is that even to test a model [Rails needs to be initialized](https://github.com/rspec/rspec-rails/blob/main/lib/generators/rspec/install/templates/spec/rails_helper.rb#L4) so you get all the paths and autoloading to work, not to mention the db configuration and all that jazz. Only true POROs will be runnable by themselves, but that's usually a tiny minority of what it's done.

@elia, that's a fair point.

> With regards with this being temporary I see it as viable for every extension, and we're already forced to use env-vars and the CI to have a matrix of combinations (e.g. if you want to test with and without active-storage). That would cover also something like solidus_checkout.

I'm not sure I understand here. Yeah, we're forced to use env vars no matter what. But I don't think that we should diverge from the [standard way of testing engines](https://guides.rubyonrails.org/v4.0/engines.html#testing-an-engines) unless there's a compelling reason. And the reason I see now is making our temporary solution easy.

Reply:
_We agreed in a slack chat to handle this separately since it doesn't affect the release, but only the development experience_